jupyter_client/manager.py

```diff
@@ -454,12 +454,21 @@ def is_alive(self):
             # we don't have a kernel
             return False
 
-    def memory_usage(self):
+    @property
+    def kernel_stats(self):
+        """
+        Returns a dict containing functions that gather
+        different stats about the kernel.
+        """
+        return {
+            "memory_usage": self._memory_usage
+        }
+
+    def _memory_usage(self):
         """
         Returns memory usage of the kernel in MB.
         Function should be overwritten if there is a
```

Comment:
This should remain a local function per its `_` prefix.  I.e., one not intended for overrides.  Per previous comment, `kernel_stats()` will be what others can override.

Reply:
I thought about maybe putting it into the `kernel_stats` function as an internal defined function,
this way it will not pollute the namespace of the class iteself, and people will know that this function is only to be used from `kernel_stats`.
Thoughts?